static/app/views/performance/newTraceDetails/traceSearch/traceSearchTokenizer.tsx

```diff
@@ -0,0 +1,124 @@
+import type {
+  NoDataNode,
+  ParentAutogroupNode,
+  SiblingAutogroupNode,
+  TraceTree,
+  TraceTreeNode,
+} from './../traceTree';
+import grammar from './traceSearch.pegjs';
+
+interface SearchToken {
+  key: string;
+  type: 'Token';
+  value: string | number;
+  negated?: boolean;
+  operator?: 'gt' | 'ge' | 'lt' | 'le' | 'eq';
+}
+
+type Token = SearchToken;
+
+// typeof can return one of the following string values - we ignore BigInt, Symbol as there
+// is no practical case for them + they are not supported by JSON.
+// object (special handling for arrays), number, string, boolean, undefined, null
+type Type = 'object' | 'number' | 'string' | 'boolean' | 'undefined' | 'null';
+
+// @ts-expect-error we ignore some keys on purpose, the TS error makes it helpful
+// for seeing exactly which ones we are ignoring for when we want to add support for them
+const SPAN_KEYS: Record<keyof TraceTree.Span, Type> = {
+  hash: 'string',
+  span_id: 'string',
+  start_timestamp: 'number',
+  timestamp: 'number',
+  trace_id: 'string',
+  description: 'string',
+  exclusive_time: 'number',
+
+  op: 'string',
+  origin: 'string',
+  parent_span_id: 'string',
+  same_process_as_parent: 'boolean',
+  // This one will need to be flattened
+  'span.average_time': 'number',
+  status: 'string',
+
+  // These are both records and will need to be handled differently
+  sentry_tags: 'string',
+  tags: 'string',
+};
+
+export function traceSearchTokenizer(input: string): Token[] {
+  return grammar.parse(input);
+}
+
+export function traceSearchLexer(_input: string): string[] {
+  throw new Error('Not implemented');
```

Comment:
I'm going to add a unimplemented lexer pass so that we can assemble an AST and support operator precedence and OR/AND expressions later on